File: Codeforces Problems/AluckyTIcket.java
import java.util.Scanner;

public class AluckyTIcket {

    static int n;
    static String str;
    static long cn = 0;

    static long helper(long[] num) {
        long sum = 0;
        for(int i = 0 ; i < num.length ; i++)
            sum += num[i];
        // System.out.println(sum);
        return sum;
    }

    static void solve()
{
    String s1 = "";
    String s2 = "";
    for (int i = 0; i < n; i++)
    {
        if (str.charAt(i) != '4' && str.charAt(i) != '7')
        {
            System.out.println("NO");
            return;
        }
    }
    for (int i = 0; i < n / 2; i++)
    {
        s1 = s1 + str.charAt(i);
    }
    // System.out.println(s1 + " " + s1.length() + " ");
    for (int i = (n / 2); i < n; i++)
    {
        s2 = s2 + str.charAt(i);
    }
    // System.out.println(s2 + " " + s2.length() + "\n");

    long[] temp = new long[(n/2)];
    long[] temp2 = new long[(n/2)];

    for(int i = 0 ; i < n / 2 ; i++)
        temp[i] = Long.parseLong(String.valueOf(s1.charAt(i)));

    // int j = n / 2 ;
    for(int i = 0 ; i < n / 2; i++){    
        temp2[i] = Long.parseLong(String.valueOf(s2.charAt(i)));
        // j++;
    }
    if (helper(temp) == helper(temp2))
    {
        System.out.println("YES");
    }
    else
    {
        System.out.println("NO");
    }

    return;
}

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        n = sc.nextInt();
        sc.nextLine();
        str = sc.nextLine();
        solve();
    }
}

File: Codeforces Problems/ALuckyTicket.cpp
#include <bits/stdc++.h>
using namespace std;
#define ll long long
#define ld long double
#define vll vector<ll>
#define vi vector<int>
#define vb vector<bool>
#define pi pair<int, int>
#define pll pair<ll, ll>
#define vp vector<pi>
#define vpll vector<pll>
#define pb push_back
#define mp make_pair
#define mt make_tuple
#define F first
#define S second
#define For(i, a, b) for (ll i = a; i < b; i++)
// #define debug2(x,y) cout<<"This side ----> "<<#x<<" -> "<<x<<" | "<<#y<<" -> "<<y<<endl;
// #define debug(x) cout<<"This side    ----> "<<#x<<" -> "<<x<<endl
#define all(x) x.begin(), x.end()
#define rall(x) x.rbegin(), x.rend()
#define mint map<int, int>
#define mall map<ll, ll>
// #define ciN cin
// #define gu(a,s) get<a>(s)
#define tin tuple<ll, ll, ll>
#define ter(x, y, z) ((x) ? y : z)
// #define ul ll

ll n;
string str;
ll cn = 0;

ll helper(ll num)
{
    ll sum = 0;
    while (num > 0)
    {
        sum = sum + (num % 10);
        num = num / 10;
    }
    cout << sum << endl;
    return sum;
}

void solve()
{
    string s1 = "";
    string s2 = "";
    for (ll i = 0; i < n; i++)
    {
        if (str.at(i) != '4' && str.at(i) != '7')
        {
            cout << "NO";
            return;
        }
    }
    for (ll i = 0; i < n / 2; i++)
    {
        s1 = s1 + str.at(i);
    }
    cout << s1 << " " << s1.size() << " ";
    for (ll i = (n / 2); i < n; i++)
    {
        s2 = s2 + str.at(i);
    }
    cout << s2 << " " << s2.size() << endl;
    stringstream obj1(s1);
    ll temp = 0;
    obj1 >> temp;

    stringstream obj2(s2);
    ll temp2;
    obj2 >> temp2;
    if (helper(temp) == helper(temp2))
    {
        cout << "YES";
    }
    else
    {
        cout << "NO";
    }

    return;
}

int main()
{
    cin >> n >> str;
    solve();
}